operator/pkg/validate/validate.go

```diff
@@ -200,6 +201,31 @@ func validateRevision(_ util.Path, val interface{}) util.Errors {
 	return nil
 }
 
+// IsRevisionString returns error if non-string revision is passed
+func IsRevisionString(iopYAML string) error {
+	var specTree = make(map[string]interface{})
+	spec, err := tpath.GetConfigSubtree(iopYAML, "spec")
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal([]byte(spec), &specTree)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling spec overlay yaml into untype tree %v", err)
+	}
+	rev := specTree["revision"]
+	// Skip if revision is not specified in any of the profile
+	if util.IsEmptyString(rev) {
+		return nil
+	}
+	_, ok := rev.(string)
+	// fail early if revision is not a string eg: revision: 18, revision: 1.8, revision: 1.8.0
+	if !ok {
+		rev = fmt.Sprintf("%v", rev)
+		return fmt.Errorf("invalid revision specified: %v. revision must be a string eg: %q", rev, "1-9-0")
```

Comment:
Nit: `revision` after the period should be capitalized.  Revision referse to the concept of a revisio", not the variable `revision` which is case sensitive.

Reply:
Done.